### PR TITLE
Create icon only button

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** #199 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

This PR reflects the creation of the icon only button, along with the necessary styles and example documentation for the new feature.

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: demo/demo.md [hunk 1]
```diff
@@ -215,6 +215,143 @@ Be sure to use the `customColor` attribute on the auro-icon component to allow c
...
 </auro-accordion>
 
+#### Icon only button
+
+Use the `iconOnly` attribute with the `auro-button` element for a style specific to having only an `auro-icon` as the button's content. 
+
+Be sure to also use the `customColor` and `customSize` attributes on the `auro-icon` component to allow colors set in your parent element to pass through to the icon.
+
+<div class="exampleWrapper auro_containedButtons">
+  <auro-button iconOnly>
```
Comment (jason-capsule42) [inline demo/demo.md:225]: Can we update all these examples to include an `aria-label` attribute for voiceover support?

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]
See my comment about aria-labels. Update that and then let's ship it!

File: package.json [hunk 2]
```diff
@@ -65,10 +65,11 @@
     "eslint": "^7.32.0",
     "eslint-plugin-lit": "^1.5.1",
     "focus-visible": "^5.2.0",
-    "husky": "^4.3.8",
+    "husky": "^8.0.0",
     "lodash": "^4.17.21",
     "marked": "^4.0.10",
     "nodemon": "^2.0.12",
+    "node-sass": "^8.0.0",
```
Comment (blackfalcon) [inline package.json:72]: This is not how we need to update to dart-sass. Please see this commit for all the details that are needed. 

https://github.com/AlaskaAirlines/auro-badge/pull/45/commits/f1edf77ba148adec56ebbd4fe0e2965ec9465977

By not updating wc-sass-render, you still have a dependency on node-sass. That's why the `node-sass` commands in the npm scripts are still working.

**Review by blackfalcon — CHANGES_REQUESTED** [review]
There are a lot of changes here that are not complete when trying to upgrade to a node 18/18 project. Karma is not compatible. Also, there are a number of packages that require node 18 to run. 

```
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@semantic-release/npm@10.0.3',
npm WARN EBADENGINE   required: { node: '>=18' },
npm WARN EBADENGINE   current: { node: 'v16.13.0', npm: '8.5.2' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'env-ci@9.0.0',
npm WARN EBADENGINE   required: { node: '^16.14 || >=18' },
npm WARN EBADENGINE   current: { node: 'v16.13.0', npm: '8.5.2' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'execa@7.1.1',
npm WARN EBADENGINE   required: { node: '^14.18.0 || ^16.14.0 || >=18.0.0' },
npm WARN EBADENGINE   current: { node: 'v16.13.0', npm: '8.5.2' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'semantic-release@21.0.1',
npm WARN EBADENGINE   required: { node: '>=18' },
npm WARN EBADENGINE   current: { node: 'v16.13.0', npm: '8.5.2' }
npm WARN EBADENGINE }
```

I'd really suggest not doing the repo updates and keeping this work restricted to the scope of the update needed. We will address all the other updates in a PR similar to https://github.com/AlaskaAirlines/auro-badge/pull/45

**Comment by jordanjones243** [comment]
Due to an error that could not be overcome within my local clone of `auro-button`, this PR will now be closed and the work in this PR will be reflected in another PR:
* https://github.com/AlaskaAirlines/auro-button/pull/202

**Pull request closed without merging**